assets/js/arrayProcessingTool.js

```diff
@@ -1,94 +1,187 @@
-                if (this.arr[i] < 0) {
-                    break;
-                }
-                if ((this.arr[i + j] < 0) && (Math.abs(this.arr[i + j]) > currentSum)) {
-                    i += j;
-                    break;
-                } else {
-                    currentSum += this.arr[i + j];
-                }
+        let currentSum = 0;
+        for (let i = 0; i < this.arr.length; i++) {
+            console.log('i++ ' + i);
+            while (this.arr[i] < 1 && i < this.arr.length) {
+                console.log('<0');
+                i++;
             }
+            currentSum = this.arr[i];
             if (currentSum > sum) {
                 sum = currentSum;
             }
-            return sum;
+            if (i === this.arr.length - 1) {
+                console.log(this.arr[i] + ' last el');
+                console.log('curS:' + currentSum, 'sum: ' + sum, 'in I');
+                break;
+            }
+            for (let j = i + 1; j < this.arr.length; j++) {
+                console.log(this.arr[i], this.arr[j], currentSum);
+                if ((this.arr[j] < 0) && (Math.abs(this.arr[j]) > currentSum)) {
+                    console.log('- Elem: ' + this.arr[j]);
+                    i = j;
+                    if (currentSum > sum) {
+                        sum = currentSum;
+                    }
+                    break;
+                } else {
+                    console.log('Суммируем');
+                    currentSum += this.arr[j];
+                }
+                console.log('curS:' + currentSum, 'sum: ' + sum, 'in J');
+                if (currentSum > sum) {
+                    sum = currentSum;
+                }
+            }
         }
+        return sum;
     },
-    search:function (){
-        const arrLength=this.arr.length;
-        if(!arrLength){
+    search: function () {
+        let arrTmp = [];
+        arrTmp = this.arr.map(el => {
+            if (!arrTmp.includes(el)) {
+                return el;
+            }
+        })
+        console.log(arrTmp);
+        const arrLength = arrTmp.length;
+        if (!arrLength) {
             console.log('array is empty');
-            return {};
+            return 'array is empty';
         }
-        if (arrLength===1){
-            return {max:this.arr[0],min:this.arr[0],middle:this.arr[0]};
+        let max = arrTmp[0], min = arrTmp[0];
+        let canFindMiddle = arrLength % 2;
+        let mid = canFindMiddle ? arrTmp[0] : '';
+        if (arrLength === 1) {
+            return 'max: ' + arrTmp[0] + ' min: ' + arrTmp[0] + ' middle:' + arrTmp[0];
         }
-        let max=this.arr[0],min=this.arr[0];
-        for (let i=1;i<arrLength;i++){
-            if (this.arr[i]>max){
-                max=this.arr[i];
+        if (arrLength === 2) {
+            if (arrTmp[1] > arrTmp[0]) {
+                return 'max: ' + arrTmp[1] + ' min: ' + arrTmp[0] + ' middle:' + mid;
```

Comment:
_Code style_: replace string formatting repetitive code. Define a function for it or make a single return in the method.

Reply:
fixed